### PR TITLE
Add CraftTweaker documentation for Summon Creatures.

### PR DESCRIPTION
**Pull request by noobanidus**
A new feature in the next release of Roots 3.

### CODE REVIEW

**Comment by noobanidus** [comment]
And I forgot completely until Jared just reminded me. I think it should be resolved correctly now.

File: docs/Mods/Roots_3/summoncreatures.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+### Class
+
+```zenscript
+import mods.roots.SummonCreatures;
+```
+
+#### Methods
+
+```zenscript
+void addEntity(
```
Comment (kindlich) [inline docs/Mods/Roots_3/summoncreatures.md]: Just out of curiosity, what happens when Ingredients match more than one recipe?
Will both be summoned?
The first to be registered? The first found in a (possibly) unordered Map?

File: docs/Mods/Roots_3/summoncreatures.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+);
+```
+
+
+---
+
+
+```zenscript
+void removeEntity(
+  IEntityDefinition entity // the entity to remove from summoning via recipe
```
Comment (kindlich) [inline docs/Mods/Roots_3/summoncreatures.md]: Does this remove all recipes for that entity?
Or does this remove the first recipe that it finds?

Also, does this remove recipes added via CrT or do you use the adding and removing actions so it only removes recipes that were not added via scripts?

File: docs/Mods/Roots_3/summoncreatures.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  IEntityDefinition entity // the entity to add life essence for
+);
+```
+
+
+---
+
+
+```zenscript
+void clearLifeEssence();
```
Comment (kindlich) [inline docs/Mods/Roots_3/summoncreatures.md]: Same question as for removing in general:
Does this also remove recipes added via script or only the ones that were present before?

File: docs/Mods/Roots_3/summoncreatures.md (hunk 2 above)
Comment (noobanidus) [reply docs/Mods/Roots_3/summoncreatures.md]: As discussed on Discord, all events occur in the order that they are given in the script file.

File: docs/Mods/Roots_3/summoncreatures.md (hunk 3 above)
Comment (noobanidus) [reply docs/Mods/Roots_3/summoncreatures.md]: See response to other comment.

File: docs/Mods/Roots_3/summoncreatures.md (hunk 1 above)
Comment (noobanidus) [reply docs/Mods/Roots_3/summoncreatures.md]: Oh, as I mentioned on Discord, it's a race condition.

**Comment by jaredlll08** [comment]
can this be merged?

**Comment by noobanidus** [comment]
I believe so, I addressed all of Kindlich's concerns over Discord and he seemed happy?

**Pull request merged**